loafwallet/Enter2FACodeView.swift

```diff
@@ -0,0 +1,154 @@
+//
+//  Enter2FACodeView.swift
+//  loafwallet
+//
+//  Created by Kerry Washington on 8/9/21.
+//  Copyright © 2021 Litecoin Foundation. All rights reserved.
+//
+
+import SwiftUI
+
+struct Enter2FACodeView<Presenting>: View where Presenting: View {
+    
+    
+    //MARK: - Combine Variables
+    @ObservedObject
+    var twoFAViewModel: Enter2FACodeViewModel
+    
+    @Binding
+    var shouldShowEnter2FAView: Bool
+    
+    @State
+    private var disableConfirmButton: Bool = true
+    
+    var presenting: Presenting
+    
+    let tokenWidth: CGFloat = 150
+    
+    var body: some View {
+        GeometryReader { (deviceSize: GeometryProxy) in
+            HStack{
+                Spacer(minLength: 5)
+                ZStack {
+                    self.presenting.disabled(shouldShowEnter2FAView)
+                    VStack {
+                        
+                        //Dismiss button
+                        Button(action: {
+                            twoFAViewModel.shouldDismissView {
+                                self.shouldShowEnter2FAView.toggle()
+                                UIApplication.shared.endEditing()
+                            }
+                            
+                        }) {
+                            Image("whiteCross")
+                                .resizable()
+                                .aspectRatio(contentMode: .fit)
+                                .frame(width: 15,
+                                       height: 15)
+                        }
+                        .frame(minWidth: 0,maxWidth: .infinity, alignment: .trailing)
+                        
+                        Text("XXXEnter Code")
+                            .font(Font(UIFont.barlowSemiBold(size: 21.0)))
+                            .padding(.bottom, 8)
+                            .foregroundColor(Color.white)
+                         
+                        Text("XXX Enter the code that was sent to your Litecoin Card account email.")
+                            .font(Font(UIFont.barlowRegular(size: 19.0)))
+                            .foregroundColor(Color.white)
+                            .padding([.leading, .trailing], 20)
+
+                        
+                        TextField("", text: $twoFAViewModel.tokenString)
```

Comment:
This is a good example of how SwiftUI works with a MVVM architecture Here the `twoFAViewModel.tokenString` is updated from a `didSet`where it also filtered the length and char type